src/certif/certifChecker.ml

```diff
@@ -3008,12 +2972,33 @@ let generate_all_proofs uid input sys =
     end
   in
   create_dir dirname;
+  let cert_inv =
+    ref
+      {
+        k = 0;
+        name = "";
+        dirname = "";
+        phi_file = "";
+        phi_lfsc_trace_file = "";
+        base = "";
+        induction = "";
+        implication = "";
+        for_system =
+          { names = names_kind2; smt2_file = ""; smt2_lfsc_trace_file = "" };
+        kind2_system =
+          { names = names_kind2; smt2_file = ""; smt2_lfsc_trace_file = "" };
+        jkind_system =
+          { names = names_jkind; smt2_file = ""; smt2_lfsc_trace_file = "" };
+        obs_system =
+          { names = names_obs; smt2_file = ""; smt2_lfsc_trace_file = "" };
+      }
+  in
+  let syms = ref [] in
 
   if not (is_fec sys) then begin
-    
     (try
-       let cert_inv = generate_split_certificates sys dirname in
-       Proof.generate_inv_proof cert_inv;
+      cert_inv := generate_split_certificates sys dirname;
+      syms := Proof.generate_inv_proof !cert_inv
      with e ->
        (* Send statistics *)
        KEvent.stat Stat.[certif_stats_title, certif_stats];
```

Comment:
I'd rather you avoided the use of `ref` for `cert_inv` and `syms`. You can achieve that by using:
```ocaml
let cert_inv, syms =
  (try
    let cert_inv = generate_split_certificates sys dirname in
    cert_inv, Proof.generate_inv_proof cert_inv
  with e ->
    (* Send statistics *)
    KEvent.stat Stat.[certif_stats_title, certif_stats];
    raise e);
in
```

Reply:
Did not know this was possible!